### ISSUE TITLE
Unable to enable "External storage support" app

### ISSUE THREAD

**Issue by Revertron** [nextcloud/server#2678]
### Steps to reproduce
1. Install NC 11
2. Try to enable External storage support
3. It says: Error while enabling app

### Expected behaviour
The app would be enabled

### Actual behaviour
It says: Error while enabling app

### Server configuration
**Operating system**: Debian 8 x86_64

**Web server:** Nginx 1.11

**Database:** MariaDB 10+

**PHP version:** 7.0.14

**Nextcloud version:** 11.0.0.10

**Updated from an older Nextcloud/ownCloud or fresh install:** Updated from 9 and 10

**Where did you install Nextcloud from:** nextcloud.com

**Signing status:**
No errors have been found.

**List of activated apps:**
<details>
<summary>App list</summary>
Enabled:
  - activity: 2.4.1
  - admin_audit: 1.1.0
  - apporder: 0.3.3
  - bookmarks: 0.9.1
  - calendar: 1.4.1
  - comments: 1.1.0
  - contacts: 1.5.2
  - dav: 1.1.1
  - direct_menu: 0.9.3
  - federatedfilesharing: 1.1.1
  - federation: 1.1.1
  - files: 1.6.1
  - files_clipboard: 0.5.0
  - files_pdfviewer: 1.0.1
  - files_retention: 1.0.1
  - files_sharing: 1.1.1
  - files_texteditor: 2.2
  - files_trashbin: 1.1.0
  - files_versions: 1.4.0
  - files_videoplayer: 1.0.0
  - firstrunwizard: 2.0
  - gallery: 16.0.0
  - logreader: 2.0.0
  - lookup_server_connector: 1.0.0
  - news: true
  - nextcloud_announcements: 1.0
  - notifications: 1.0.1
  - ownnote: 1.08
  - password_policy: 1.1.0
  - previewgenerator: 1.0.1
  - provisioning_api: 1.1.0
  - serverinfo: 1.1.1
  - sharebymail: 1.0.1
  - sharerenamer: 1.1
  - spreed: 1.1.2
  - survey_client: 0.1.5
  - systemtags: 1.1.3
  - templateeditor: 0.2
  - theming: 1.1.1
  - twofactor_backupcodes: 1.0.0
  - updatenotification: 1.1.1
  - workflowengine: 1.1.1
</details>

**The content of config/config.php:**
<details>
<summary>Config report</summary>

```
{
    "system": {
        "instanceid": "oc4p0q4g0czk",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "trusted_domains": [
            "cloud.jamm.me"
        ],
        "datadirectory": "\/nextcloud_data",
        "overwrite.cli.url": "https:\/\/cloud.jamm.me",
        "dbtype": "mysql",
        "version": "11.0.0.10",
        "dbname": "nextcloud",
        "dbhost": "127.0.0.1",
        "dbtableprefix": "oc_",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "installed": true,
        "loglevel": 0,
        "theme": "reverse_one",
        "remember_login_cookie_lifetime": 604800,
        "session_lifetime": 259200,
        "session_keepalive": true,
        "maintenance": false,
        "logfile": "\/nextcloud_data\/nextcloud.log",
        "appstore.experimental.enabled": true,
        "debug": false,
        "trashbin_retention_obligation": "auto",
        "memcache.local": "\\OC\\Memcache\\Memcached",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "mail_from_address": "cloud",
        "mail_smtpmode": "php",
        "mail_domain": "jamm.me",
        "updater.release.channel": "stable",
        "updater.secret": "***REMOVED SENSITIVE VALUE***"
    }
}
```
</details>

**Are you using external storage, if yes which one:** I can't because of this problem

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** No

### Client configuration
**Browser:**
50.0.2
**Operating system:**
Win 7 x86_64
#### Nextcloud log (data/nextcloud.log)
<details>
<summary>Nextcloud log</summary>
```
{"reqId":"ui3aRT\/\/bCmDxYljoFRE","remoteAddr":"1.2.3.4","app":"index","message":"Exception: {\"Exception\":\"TypeError\",\"Message\":\"Argument 1 passed to OC\\\\App\\\\DependencyAnalyzer::analyze() must be of the type array, null given, called in \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/app.php on line 1339\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/app.php(1339): OC\\\\App\\\\DependencyAnalyzer->analyze(NULL)\\n#1 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/app.php(385): OC_App::checkAppDependencies(Object(OC\\\\AllConfig), Object(OC\\\\L10N\\\\L10N), NULL)\\n#2 \\\/var\\\/www\\\/nextcloud\\\/settings\\\/ajax\\\/enableapp.php(44): OC_App->enable('files_external', NULL)\\n#3 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Route\\\/Route.php(155) : runtime-created function(1): require_once('\\\/var\\\/www\\\/nextcl...')\\n#4 [internal function]: __lambda_func()\\n#5 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/Route\\\/Router.php(299): call_user_func('\\\\x00lambda_448', Array)\\n#6 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/base.php(1010): OC\\\\Route\\\\Router->match('\\\/settings\\\/ajax\\\/...')\\n#7 \\\/var\\\/www\\\/nextcloud\\\/index.php(40): OC::handleRequest()\\n#8 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/App\\\/DependencyAnalyzer.php\",\"Line\":55}","level":3,"time":"2016-12-14T17:14:31+00:00","method":"POST","url":"\/index.php\/settings\/ajax\/enableapp.php","user":"Revertron","version":"11.0.0.10"}
```
</details>

<details>
<summary>And from another try</summary>
```
{"reqId":"x+AXvYKK3tUitEgFvbim","remoteAddr":"1.2.3.4","app":"no app in context","message":"Exception: {\"Exception\":\"Error\",\"Message\":\"Class 'OCA\\\\Files_External\\\\AppInfo\\\\Application' not found\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/app.php(214): require_once()\\n#1 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/app.php(155): OC_App::requireAppFile('files_external')\\n#2 \\\/var\\\/www\\\/nextcloud\\\/lib\\\/private\\\/legacy\\\/app.php(125): OC_App::loadApp('files_external')\\n#3 \\\/var\\\/www\\\/nextcloud\\\/ocs\\\/v1.php(56): OC_App::loadApps()\\n#4 \\\/var\\\/www\\\/nextcloud\\\/ocs\\\/v2.php(23): require_once('\\\/var\\\/www\\\/nextcl...')\\n#5 {main}\",\"File\":\"\\\/var\\\/www\\\/nextcloud\\\/apps\\\/files_external\\\/appinfo\\\/app.php\",\"Line\":35}","level":3,"time":"2016-12-14T17:42:29+00:00","method":"GET","url":"\/ocs\/v2.php\/apps\/notifications\/api\/v2\/notifications","user":"Revertron","version":"11.0.0.10"}
```
</details>

**Comment by nickvergessen** [comment]
@icewind1991

**Comment by Revertron** [comment]
Any news on this?

**Comment by nickvergessen** [comment]
Are you sure you updated the code of the app as well?

**Comment by Revertron** [comment]
Just redownloaded the NC archive, deleted apps/files_external on the server, extracted whole 'files_external' directory to it's place, adjusted owner and tried to enable.
No luck again. The first time the button becomed "Enabling app..." for 10-15 seconds, and then becomed "Disable".
Now, if I try again, it becomes "Enabling app..." on 2 seconds and then shown "Error while enabling app" message.

**Comment by rkantos** [comment]
Getting same problems while trying to enable external authentication app! I just have a fresh install of NC11.0.1.  Do you @Revertron have the same problem if trying to install anything else? Nothing I can find in the logs that would make sense for this kind of error..

**Comment by Grimeton** [comment]
I figured it out.

At least in my case memcache broke it. At first I figured maybe memcache has old data stored. So I restarted it and reenabled the files_external app again. Still the same problem.

So I downloaded a clean nextcloud 11.0.1 and started to test. I had another app in mind that breaks something but even with only the default apps enabled it didn't work. So then I thought about memcache. 

Just disabling memcache in the config doesn't help. This is what I did:

- mv old nextloud out of the way on the filesystem (nc -> nc_old)
- keep the database in place
- download clean nextcloud and put it in place in the "nc" folder
- copy the config over from the old version
- edit the config and disable memcached in the config
- go to the website, login, enable external storage and enjoy

If you get errors when enabling external storage connect to your database and drop the PREFIX_external_* tables. Also delete all values from PREFIX_appconfig where the appid is "files_external".

After that enable it again.

I now tested it with redis and redis does not break it. So I guess for now redis is the way to go.

Cu
